src/d3a/models/area/redis_external_market_connection.py

```diff
@@ -104,7 +101,10 @@ def event_market_cycle(self):
             return
         market_event_channel = f"{self.channel_prefix}/market-events/market"
         current_market_info = self.area.current_market.info
-        current_market_info['last_market_stats'] = \
+        current_market_info["current_market_fee"] = \
+            self.area.current_market.fee_class.grid_fee_rate
+        current_market_info["next_market_fee"] = str(self.area.get_grid_fee())
```

Comment:
For the next market, you are checking the grid fee type and then returning it whereas in `current_market_fee ` you are only returning the constant grid fee. Better to have a uniform implementation for both `current_market_fee  & next_market_fee ` .

Reply:
@faizan2590 not really, for the `current_market_fee` i return the grid_fee that is set, independent of the type. Have a look in the current_market.fee_class.

Reply:
Ah yes got it, again why didnt you followed the same approach for getting the `next_market_fee ` too ?

Reply:
No need for the further change: https://gridsingularity.atlassian.net/browse/D3ASIM-2827?focusedCommentId=20174